src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/DataValidationWorker.cs

```diff
@@ -0,0 +1,347 @@
+            MigrationSettings migrationSettings,
+            WorkerInput workerInput,
+            string traceId)
+            : base(stateProvider, workerInput, traceId)
+        {
+            this.initParams = this.StateProvider.GetInitParams();
+            this.servicePartitionClient = servicePartitionClient;
+            this.migrationSettings = migrationSettings;
+        }
+
+        public ITransaction Transaction { get => this.StateProvider.GetStateManager().CreateTransaction(); }
+
+        public override async Task<WorkerResult> StartWorkAsync(CancellationToken cancellationToken)
+        {
+            ActorTrace.Source.WriteInfoWithId(
+                        TraceType,
+                        this.TraceId,
+                        $"#{this.Input.WorkerId}: Starting or resuming data validation worker\n Input: {this.Input.ToString()}");
+
+            try
+            {
+                if (this.migrationSettings.PercentageOfMigratedDataToValidate > 0 && this.Input.EndSeqNum > 0)
+                {
+                    var startSN = this.Input.StartSeqNum;
+                    var endSN = this.Input.EndSeqNum;
+                    var allKeysToValidateCount = endSN - startSN + 1;
+
+                    ActorTrace.Source.WriteNoiseWithId(
+                        TraceType,
+                        this.TraceId,
+                        $"#{this.Input.WorkerId}: StartSeqNum: {startSN} EndSeqNum: {endSN}");
+
+                    // Fetch saved migrated keys
+                    var migratedKeys = new List<string>();
+                    using (var tx = this.Transaction)
+                    {
+                        var migrationKeysMigratedChunksCount = await this.MetadataDict.TryGetValueAsync(tx, MigrationConstants.MigrationKeysMigratedChunksCount);
+                        long chunksCount = 0L;
+                        if (migrationKeysMigratedChunksCount.HasValue)
+                        {
+                            chunksCount = MigrationUtility.ParseLong(migrationKeysMigratedChunksCount.Value, this.TraceId);
+
+                            ActorTrace.Source.WriteNoiseWithId(
+                                TraceType,
+                                this.TraceId,
+                                $"#{this.Input.WorkerId}: chunksCount: {chunksCount}");
+
+                            for (int i = 1; i <= chunksCount; i++)
+                            {
+                                var savedMigratedKeys = await this.MetadataDict.TryGetValueAsync(tx, Key(MigrationConstants.MigrationKeysMigrated, i));
+                                if (savedMigratedKeys.HasValue)
+                                {
+                                    migratedKeys.AddRange(savedMigratedKeys.Value.Split(DefaultDelimiter).ToList());
+                                }
+                            }
+
+                            ActorTrace.Source.WriteNoiseWithId(
```

Comment:
Info level logging would be helpful. Also these logs would not be too many, so info logging should be alright

Reply:
I meant all the logs in this method could be info level

Reply:
done